// nav-frontend-skjema/lib/checkbox.js
"use strict";

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

Object.defineProperty(exports, "__esModule", { value: true });
var PT = require("prop-types");
var React = require("react");
var classNames = require("classnames");
var skjemaelement_feilmelding_1 = require("./skjemaelement-feilmelding");
var nav_frontend_js_utils_1 = require("nav-frontend-js-utils");
var _1 = require("./");
require("nav-frontend-skjema-style");

var CheckboxGruppe = function (_React$Component) {
    _inherits(CheckboxGruppe, _React$Component);

    function CheckboxGruppe() {
        _classCallCheck(this, CheckboxGruppe);

        return _possibleConstructorReturn(this, (CheckboxGruppe.__proto__ || Object.getPrototypeOf(CheckboxGruppe)).apply(this, arguments));
    }

    _createClass(CheckboxGruppe, [{
        key: "render",
        value: function render() {
            var _props = this.props,
                children = _props.children,
                className = _props.className,
                other = _objectWithoutProperties(_props, ["children", "className"]);

            return React.createElement(_1.SkjemaGruppe, Object.assign({ className: classNames('checkboxgruppe', className) }, other), children);
        }
    }]);

    return CheckboxGruppe;
}(React.Component);

exports.CheckboxGruppe = CheckboxGruppe;
var cls = function cls(className) {
    return classNames('skjemaelement', className);
};
var inputCls = function inputCls(harFeil) {
    return classNames('skjemaelement__input checkboks', {
        'skjemaelement__input--harFeil': harFeil
    });
};
/**
 * Checkbox
 */

var Checkbox = function (_React$Component2) {
    _inherits(Checkbox, _React$Component2);

    function Checkbox() {
        _classCallCheck(this, Checkbox);

        return _possibleConstructorReturn(this, (Checkbox.__proto__ || Object.getPrototypeOf(Checkbox)).apply(this, arguments));
    }

    _createClass(Checkbox, [{
        key: "render",
        value: function render() {
            var _props2 = this.props,
                className = _props2.className,
                label = _props2.label,
                id = _props2.id,
                feil = _props2.feil,
                checkboxRef = _props2.checkboxRef,
                other = _objectWithoutProperties(_props2, ["className", "label", "id", "feil", "checkboxRef"]);

            var inputId = id || nav_frontend_js_utils_1.guid();
            var feilmeldingId = nav_frontend_js_utils_1.guid();
            // tslint:disable:react-a11y-role-has-required-aria-props
            return React.createElement(_1.SkjemaGruppeFeilContext.Consumer, null, function (context) {
                var feilmelding = context.feil || feil;
                var feilmeldingId = context.feilmeldingId || nav_frontend_js_utils_1.guid();
                return React.createElement("div", { className: cls(className) }, React.createElement("input", Object.assign({ type: "checkbox", className: inputCls(feil), id: inputId, ref: checkboxRef, "aria-invalid": !!feilmelding, "aria-errormessage": feilmelding ? feilmeldingId : undefined }, other)), React.createElement("label", { className: "skjemaelement__label", htmlFor: inputId }, label), !!!context.feil && !!feil && React.createElement(skjemaelement_feilmelding_1.default, { id: feilmeldingId }, typeof feilmelding !== 'boolean' && feilmelding));
            });
        }
    }]);

    return Checkbox;
}(React.Component);

exports.Checkbox = Checkbox;
Checkbox.propTypes = {
    /**
     * ClassName
     */
    className: PT.string,
    /**
     * Label for checkbox
     */
    label: PT.node.isRequired,
    /**
     * Id for checkbox, hvis id ikke er satt brukes en tilfeldig guid
     */
    id: PT.string,
    /**
     * Hvis skjemaet har feil sender man inn et objekt med en feilmelding
     */
    feil: PT.oneOfType([PT.node, PT.bool]),
    /**
     * Referanse til selve checkboxen. Brukes for eksempel til å sette fokus
     */
    checkboxRef: PT.func
};
Checkbox.defaultProps = {
    className: undefined,
    id: undefined,
    feil: undefined,
    checkboxRef: undefined
};
exports.default = Checkbox;
CheckboxGruppe.__docgenInfo = {"description":"","displayName":"CheckboxGruppe","methods":[],"props":{"children":{"defaultValue":null,"description":"En eller flere children, oftest en eller flere .skjemaelement","name":"children","parent":{"fileName":"nav-frontend-moduler/packages/node_modules/nav-frontend-skjema/src/skjema-gruppe.tsx","name":"SkjemaGruppeProps"},"required":true,"type":{"name":"string | number | boolean | {} | ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)>) | (new (props: any) => Component<...>)> | ... 47 more ... | (ReactNode[] & ReactPortal)"}},"className":{"defaultValue":null,"description":"ClassName","name":"className","parent":{"fileName":"nav-frontend-moduler/packages/node_modules/nav-frontend-skjema/src/skjema-gruppe.tsx","name":"SkjemaGruppeProps"},"required":false,"type":{"name":"string"}},"tag":{"defaultValue":null,"description":"Bestemmer hvilken HTML-tag som skal brukes. Default er 'fieldset'.","name":"tag","parent":{"fileName":"nav-frontend-moduler/packages/node_modules/nav-frontend-skjema/src/skjema-gruppe.tsx","name":"SkjemaGruppeProps"},"required":false,"type":{"name":"string"}},"legend":{"defaultValue":null,"description":"Legend/tittel for skjemagruppen","name":"legend","parent":{"fileName":"nav-frontend-moduler/packages/node_modules/nav-frontend-skjema/src/skjema-gruppe.tsx","name":"SkjemaGruppeProps"},"required":false,"type":{"name":"ReactNode"}},"feil":{"defaultValue":null,"description":"Hvis skjemagruppen har feil sender man inn et objekt med en feilmelding","name":"feil","parent":{"fileName":"nav-frontend-moduler/packages/node_modules/nav-frontend-skjema/src/skjema-gruppe.tsx","name":"SkjemaGruppeProps"},"required":false,"type":{"name":"ReactNode"}},"feilmeldingId":{"defaultValue":null,"description":"Valgfri ID til intern SkjemaelementFeilmelding","name":"feilmeldingId","parent":{"fileName":"nav-frontend-moduler/packages/node_modules/nav-frontend-skjema/src/skjema-gruppe.tsx","name":"SkjemaGruppeProps"},"required":false,"type":{"name":"string"}}}}
